src/Payments/Integrations/Stripe.php

```diff
@@ -189,11 +189,14 @@ private function get_button_locale() {
 	 * @param PaymentResult  $result  Result object for the payment.
 	 */
 	public function add_payment_request_order_meta( PaymentContext $context, PaymentResult $result ) {
-		// phpcs:ignore WordPress.Security.NonceVerification
-		$post_data = $_POST;
-		$_POST     = $context->payment_data;
-
-		WC_Stripe_Payment_Request::add_order_meta( $context->order->id, $context->payment_data );
-		$_POST = $post_data;
+		$data = $context->payment_data;
+		if ( $data['payment_request_type'] && 'stripe' === $context->payment_method ) {
+			// phpcs:ignore WordPress.Security.NonceVerification
+			$post_data = $_POST;
+			$_POST     = $context->payment_data;
+			$context->set_payment_method( 'stripe' );
```

Comment:
You already checked this in the parent conditional - it was already `stripe`

Reply:
oops, leftover from a previous iteration of this pull where I was trying to just detect if `payment_request` was the `payment_method` value and then switch to `stripe` (which didn't work because payment_method is validated earlier than the hook (for good reason).